sdk/program/src/wasm/mod.rs

```diff
@@ -3,7 +3,8 @@
 use wasm_bindgen::prelude::*;
 
 pub mod instructions;
-pub mod system_instruction;
+#[deprecated(since = "2.2.0", note = "Use `solana_system_interface` crate instead")]
+pub use solana_system_interface::wasm as system_instruction;
```

Comment:
`solana_system_interface::wasm` isn't meant to be imported as there is nothing in it that can be imported. It's just an `impl` block for the SystemInstruction struct. This is also true for `solana_program::wasm::system_instruction`.

I would just put an empty module here to avoid a breaking change (only a breaking change for people who were pointlessly importing the wasm mod).